packages/liveblocks-core/src/ai.ts

```diff
@@ -524,32 +531,42 @@ function createStore_forChatMessages(
       // - and only if the current client ID is the designated client ID
       //
       if (message.role === "assistant" && message.status === "awaiting-tool") {
-        for (const toolCall of message.contentSoFar.filter(
+        for (const toolInvocation of message.contentSoFar.filter(
           (part) =>
             part.type === "tool-invocation" && part.stage === "executing"
         )) {
-          if (seenToolCallIds.has(toolCall.invocationId)) {
+          if (seenToolInvocationIds.has(toolInvocation.invocationId)) {
             // Do nothing, we already know of it
             continue;
           }
 
-          seenToolCallIds.add(toolCall.invocationId);
+          seenToolInvocationIds.add(toolInvocation.invocationId);
 
           const toolDef = toolsStore
-            .getToolΣ(toolCall.name, message.chatId)
+            .getToolΣ(toolInvocation.name, message.chatId)
             .get();
 
           const respondSync = <R extends JsonObject>(
             ...args: OptionalTupleUnless<R, [result: ToolResultResponse<R>]>
           ) => {
             const [result] = args;
+            if (executedToolInvocationIds.has(toolInvocation.invocationId)) {
+              console.warn(
+                `Ignoring respond(): tool '${toolInvocation.name}' (${toolInvocation.invocationId}) has already executed`
```

Comment:
Actually, this isn't the correct fix. Please ignore, will update the PR with the proper fix.